src/tree_data.c

```diff
@@ -3267,3 +3272,155 @@ lyd_get_value(const struct lyd_node *node)
 
     return NULL;
 }
+
+/**
+ * @brief Gets the term data node extension record for given node
+ *
+ * @param[in] node The term data node.
+ * @param[out] record The term data node extension record
+ * @param[in] create Whether to create record if not exists
+ * @return LY_SUCCESS on success.
+ * @return LY_ERR value on error.
+ */
+LY_ERR
+lyd_get_or_create_term_nodes_ext_record(const struct lyd_node_term *node, struct lyd_term_nodes_ext_rec **record, ly_bool create)
```

Comment:
I believe this function can be `static`.

Reply:
Done